Investigate and fix failing GridProperties test

```sh
FAILED tests/test_grid3d/test_grid_properties.py::test_gridproperties_from_roff - ValueError: Requested keyword NA<< is not in ROFF file, valid entries are [], set strict=False to warn instead.
```

This test fails when hypothesis generates `NA<<` as a name. It seems to be fixed by adjusting `min_codepoint=60` to 65 or some value that excludes the `<` character (where 60 is the ASCII decimal value). 

https://github.com/equinor/xtgeo/blob/9bb3ccf4df4fea4de632317965b5c3c034631d64/tests/test_grid3d/gridprop_generator.py#L10-L12

Should double check that this actually isn't a bug being caught, i.e. `<` is a valid keyword/name character (but it's probably not).

--- a/src/xtgeo/grid3d/_grid3d_utils.py
+++ b/src/xtgeo/grid3d/_grid3d_utils.py
@@ -1,7 +1,9 @@
-# -*- coding: utf-8 -*-
-
 """Some grid utilities, file scanning etc (methods with no class)"""
+
+from __future__ import annotations
+
 import re
+from typing import TYPE_CHECKING, List, Literal, Union
 
 import pandas as pd
 
@@ -10,13 +12,22 @@ import xtgeo.cxtgeo._cxtgeo as _cxtgeo
 from xtgeo import XTGeoCLibError
 from xtgeo.common.constants import MAXDATES, MAXKEYWORDS
 
+if TYPE_CHECKING:
+    from xtgeo.common.sys import _XTGeoFile
+
+    from .grid_properties import GridPropertiesKeywords, KeywordDateTuple, KeywordTuple
+
 xtg = xtgeo.XTGeoDialog()
 logger = xtg.functionlogger(__name__)
 
 
 def scan_keywords(
-    pfile, fformat="xecl", maxkeys=MAXKEYWORDS, dataframe=False, dates=False
-):
+    pfile: _XTGeoFile,
+    fformat: Literal["xecl", "roff"] = "xecl",
+    maxkeys: int = MAXKEYWORDS,
+    dataframe: bool = False,
+    dates: bool = False,
+) -> GridPropertiesKeywords:
     """Quick scan of keywords in Eclipse binary restart/init/... file,
     or ROFF binary files.
 
@@ -27,21 +38,20 @@ def scan_keywords(
 
     if fformat == "xecl":
         if dates:
-            data = _scan_ecl_keywords_w_dates(
+            keywords = _scan_ecl_keywords_w_dates(
                 pfile, maxkeys=maxkeys, dataframe=dataframe
             )
         else:
-            data = _scan_ecl_keywords(pfile, maxkeys=maxkeys, dataframe=dataframe)
-
+            keywords = _scan_ecl_keywords(pfile, maxkeys=maxkeys, dataframe=dataframe)
+    elif fformat == "roff":
+        keywords = _scan_roff_keywords(pfile, maxkeys=maxkeys, dataframe=dataframe)
     else:
-        data = _scan_roff_keywords(pfile, maxkeys=maxkeys, dataframe=dataframe)
-
+        raise ValueError(f"File format can be either `roff` or `xecl`, given {fformat}")
     pfile.cfclose()
+    return keywords
 
-    return data
 
-
-def scan_dates(pfile, maxdates=MAXDATES, dataframe=False):
+def scan_dates(pfile: _XTGeoFile, maxdates: int = MAXDATES, dataframe: bool = False):
     """Quick scan dates in a simulation restart file.
 
     Cf. grid_properties.py description
@@ -81,11 +91,12 @@ def scan_dates(pfile, maxdates=MAXDATES, dataframe=False):
     return zdates
 
 
-def _scan_ecl_keywords(pfile, maxkeys=MAXKEYWORDS, dataframe=False):
-    logger.info("Scanning ECL keywords...")
+def _scan_ecl_keywords(
+    pfile: _XTGeoFile, maxkeys: int = MAXKEYWORDS, dataframe: bool = False
+) -> Union[List[KeywordTuple], pd.DataFrame]:
     cfhandle = pfile.get_cfhandle()
 
-    # maxkeys*10 is used for 1D kewords; 10 => max 8 letters in eclipse +
+    # maxkeys*10 is used for 1D keywords; 10 => max 8 letters in eclipse +
     # "|" + "extra buffer"
     nkeys, keywords, rectypes, reclens, recstarts = _cxtgeo.grd3d_scan_eclbinary(
         cfhandle, maxkeys * 10, maxkeys, maxkeys, maxkeys
@@ -131,12 +142,11 @@ def _scan_ecl_keywords(pfile, maxkeys=MAXKEYWORDS, dataframe=False):
     return result
 
 
-def _scan_ecl_keywords_w_dates(pfile, maxkeys=MAXKEYWORDS, dataframe=False):
+def _scan_ecl_keywords_w_dates(
+    pfile: _XTGeoFile, maxkeys: int = MAXKEYWORDS, dataframe: bool = False
+) -> Union[List[KeywordDateTuple], pd.DataFrame]:
     """Add a date column to the keyword"""
-
-    logger.info("Scan keywords with dates...")
     xkeys = _scan_ecl_keywords(pfile, maxkeys=maxkeys, dataframe=False)
-
     xdates = scan_dates(pfile, maxdates=MAXDATES, dataframe=False)
 
     result = []
@@ -160,7 +170,9 @@ def _scan_ecl_keywords_w_dates(pfile, maxkeys=MAXKEYWORDS, dataframe=False):
     return result
 
 
-def _scan_roff_keywords(pfile, maxkeys=MAXKEYWORDS, dataframe=False):
+def _scan_roff_keywords(
+    pfile: _XTGeoFile, maxkeys: int = MAXKEYWORDS, dataframe: bool = False
+) -> Union[List[KeywordTuple], pd.DataFrame]:
     rectypes = _cxtgeo.new_intarray(maxkeys)
     reclens = _cxtgeo.new_longarray(maxkeys)
     recstarts = _cxtgeo.new_longarray(maxkeys)

--- a/src/xtgeo/grid3d/_gridprops_import_roff.py
+++ b/src/xtgeo/grid3d/_gridprops_import_roff.py
@@ -1,16 +1,19 @@
-from typing import List, Union
+from __future__ import annotations
 
-from typing_extensions import Literal
+import io
+from typing import TYPE_CHECKING, List, Literal, Union
+
+import roffio
 
 import xtgeo
-from xtgeo.common.sys import _XTGeoFile
+from xtgeo.common import XTGeoDialog
 
-from . import _grid3d_utils as utils
-from .grid_property import GridProperty
+if TYPE_CHECKING:
+    from xtgeo.common.sys import _XTGeoFile
 
-xtg = xtgeo.common.XTGeoDialog()
+    from .grid_property import GridProperty
 
-logger = xtg.functionlogger(__name__)
+logger = XTGeoDialog().functionlogger(__name__)
 
 
 def import_roff_gridproperties(
@@ -18,48 +21,55 @@ def import_roff_gridproperties(
     names: Union[List[str], Literal["all"]],
     strict: bool = True,
 ) -> List[GridProperty]:
-    """Imports list of properties from a roff file.
+    """
+    Imports a list of properties from a ROFF file.
 
-    Args:
-        pfile: Reference to the file
-        names: List of names to fetch, can also be "all" to fetch all properties.
-        strict: If strict=True, will raise error if key is not found.
+    Parameters:
+        pfile:
+            Reference to the file.
+        names:
+            List of names to fetch, can also be "all" to fetch all properties.
+        strict:
+            If strict=True, will raise error if key is not found.
+            Defaults to True.
+
     Returns:
         List of GridProperty objects fetched from the ROFF file.
+
     """
-    print(pfile)
-    validnames = []
+    validnames = set()
+    with roffio.lazy_read(pfile.file) as contents:
+        for tagname, tagkeys in contents:
+            for keyname, values in tagkeys:
+                if tagname == "parameter" and keyname == "name":
+                    validnames.add(values)
 
-    collectdata = utils.scan_keywords(pfile, fformat="roff")
-    for item in collectdata:
-        keyname = item[0]
-        if keyname.startswith("parameter!name!"):
-            # format is 'parameter!name!FIPNUM'
-            validnames.append(keyname.split("!").pop())
+    # Rewind if this file is in memory
+    if isinstance(pfile.file, (io.BytesIO, io.StringIO)):
+        pfile.file.seek(0)
 
-    usenames = []
+    usenames = set()
     if names == "all":
         usenames = validnames
     else:
         for name in names:
-            if name not in validnames:
-                if strict:
-                    raise ValueError(
-                        f"Requested keyword {name} is not in ROFF file, valid "
-                        f"entries are {validnames}, set strict=False to warn instead."
-                    )
-                else:
-                    logger.warning(
-                        "Requested keyword %s is not in ROFF file. Entry will"
-                        "not be read, set strict=True to raise Error instead.",
-                        name,
-                    )
-            else:
-                usenames.append(name)
+            if name in validnames:
+                usenames.add(name)
+                continue
+
+            if strict:
+                raise ValueError(
+                    f"Requested keyword {name} is not in ROFF file, valid "
+                    f"entries are {validnames}, set strict=False to warn instead."
+                )
+            logger.warning(
+                "Requested keyword %s is not in ROFF file. Entry will"
+                "not be read, set strict=True to raise Error instead.",
+                name,
+            )
 
     props = [
         xtgeo.gridproperty_from_file(pfile.file, fformat="roff", name=name)
         for name in usenames
     ]
-
     return props

--- a/src/xtgeo/grid3d/grid_properties.py
+++ b/src/xtgeo/grid3d/grid_properties.py
@@ -1,9 +1,11 @@
-# -*- coding: utf-8 -*-
-
 """Module for Grid Properties."""
+from __future__ import annotations
+
 import hashlib
+import io
+import pathlib
 import warnings
-from typing import List, Optional
+from typing import List, Literal, Optional, Tuple, Union
 
 import deprecation
 import numpy as np
@@ -20,6 +22,12 @@ from .grid_property import GridProperty
 
 xtg = XTGeoDialog()
 logger = xtg.functionlogger(__name__)
+
+KeywordTuple = Tuple[str, str, int, int]
+KeywordDateTuple = Tuple[str, str, int, int, Union[str, int]]
+GridPropertiesKeywords = Union[
+    List[Union[KeywordTuple, KeywordDateTuple]], pd.DataFrame
+]
 
 
 def gridproperties_from_file(
@@ -291,7 +299,6 @@ class GridProperties(_Grid3D):
         # compatability until the names setter has been
         # deprecated
         self._names = []
-
         self.props = props or []
 
     def __repr__(self):  # noqa: D105
@@ -724,44 +731,56 @@ class GridProperties(_Grid3D):
 
     @staticmethod
     def scan_keywords(
-        pfile, fformat="xecl", maxkeys=MAXKEYWORDS, dataframe=False, dates=False
-    ):
+        pfile: Union[str, pathlib.Path, io.BytesIO, io.StringIO],
+        fformat: Literal["roff", "xecl"] = "xecl",
+        maxkeys: int = MAXKEYWORDS,
+        dataframe: bool = False,
+        dates: bool = False,
+    ) -> GridPropertiesKeywords:
         """Quick scan of keywords in Eclipse binary files, or ROFF binary files.
 
         For Eclipse files:
-        Returns a list of tuples (or dataframe), e.g. ('PRESSURE',
-        'REAL', 355299, 3582700), where (keyword, type, no_of_values,
-        byteposition_in_file)
+            Returns a list of tuples (or dataframe), e.g.
+               ``('PRESSURE', 'REAL', 355299, 3582700)``
+            where
+                ``(keyword, type, no_of_values, byteposition_in_file)``
 
-        For ROFF files
-        Returns a list of tuples (or dataframe), e.g.
-        ('translate!xoffset', 'float', 1, 3582700),
-        where (keyword, type, no_of_values, byteposition_in_file).
+        For ROFF files:
+            Returns a list of tuples (or dataframe), e.g.
+                ``('translate!xoffset', 'float', 1, 3582700)``
+            where
+                ``(keyword, type, no_of_values, byteposition_in_file)``
 
         For Eclipse, the byteposition is to the KEYWORD, while for ROFF
         the byte position is to the beginning of the actual data.
 
-        Args:
-            pfile (str): Name or a filehandle to file with properties
-            fformat (str): xecl (Eclipse INIT, RESTART, ...) or roff for
-                ROFF binary,
-            maxkeys (int): Maximum number of keys
-            dataframe (bool): If True, return a Pandas dataframe instead
-            dates (bool): if True, the date is the last column (only
-                menaingful for restart files). Default is False.
+        Parameters:
+            pfile:
+                Name or a filehandle to file with properties.
+            fformat:
+                xecl (Eclipse INIT, RESTART, ...) or roff for ROFF binary.
+                Default is "xecl".
+            maxkeys:
+                Maximum number of keys. Default is
+                ``xtgeo.commom.constants.MAXKEYWORDS``.
+            dataframe:
+                If True, return a Pandas dataframe instead. Default is False.
+            dates:
+                If True, the date is the last column (only
+                meaningful for restart files). Default is False.
 
-        Return:
+        Returns:
             A list of tuples or dataframe with keyword info
 
         Example::
             >>> dlist = GridProperties.scan_keywords(reek_dir + "/REEK.UNRST")
 
         """
-        pfile = xtgeo._XTGeoFile(pfile)
-        pfile.check_file(raiseerror=ValueError)
+        xtg_file = xtgeo._XTGeoFile(pfile)
+        xtg_file.check_file(raiseerror=ValueError)
 
         return utils.scan_keywords(
-            pfile,
+            xtg_file,
             fformat=fformat,
             maxkeys=maxkeys,
             dataframe=dataframe,

--- a/tests/test_grid3d/test_grid_properties.py
+++ b/tests/test_grid3d/test_grid_properties.py
@@ -122,6 +122,18 @@ def test_gridproperties_from_roff(grid_property):
     assert props.names == [grid_property.name]
 
 
+def test_gridproperties_from_roff_with_name_starting_with_na():
+    grid_property = xtgeo.GridProperty(name="NA")
+    buff = io.BytesIO()
+    grid_property.to_file(buff, fformat="roff")
+    buff.seek(0)
+    props = xtgeo.gridproperties_from_file(
+        buff, fformat="roff", names=[grid_property.name]
+    )
+
+    assert props.names == [grid_property.name]
+
+
 @given(gridproperties_elements())
 def test_gridproperties_invalid_format(grid_property):
     buff = io.BytesIO()

